psd-web/app/controllers/investigations/risk_level_controller.rb

```
@@ -0,0 +1,25 @@
+module Investigations
+  class RiskLevelController < ApplicationController
+    def show
+      @investigation = Investigation.find_by!(pretty_id: params.require(:investigation_pretty_id)).decorate
+      authorize @investigation, :update?
+      @risk_level_form = RiskLevelForm.new(risk_level: @investigation.risk_level)
+    end
+
+    def update
+      @investigation = Investigation.find_by!(pretty_id: params.require(:investigation_pretty_id)).decorate
+      authorize @investigation, :update?
+      @risk_level_form = RiskLevelForm.new(params.require(:investigation).permit(:risk_level, :risk_level_other))
+      return render :show unless @risk_level_form.valid?
+
+      result = ChangeCaseRiskLevel.call(investigation: @investigation, user: current_user, risk_level: @risk_level_form.risk_level.presence)
+      if result.success?
+        flash[:success] = I18n.t(".success", scope: "investigations.risk_level", action: result.change_action, case_type: @investigation.case_type) if result.change_action.present?
+        redirect_to investigation_path(@investigation)
+      else
+        flash[:error] = I18n.t(".error", scope: "investigations.risk_level", action: result.change_action)
+        render :show
```

Could we ✂️ this `else` condition? I think in theory the service call should always be successful, as you've already checked the form is valid?

---

If you do this, I would consider using `call!` instead of `call`, as failure will be suppressed otherwise. Although it should theoretically not fail, this will at least send an exception via Sentry and alert the user something went wrong.